core/src/kmx/kmx_plus.cpp

```diff
@@ -1193,38 +1259,39 @@ kmx_plus::kmx_plus(const COMP_KEYBOARD *keyboard, size_t length)
   }
 
   const uint8_t* rawdata = reinterpret_cast<const uint8_t*>(keyboard);
-
-  sect = section_from_bytes<COMP_KMXPLUS_SECT>(rawdata+ex->kmxplus.dpKMXPlus, ex->kmxplus.dwKMXPlusSize);
+  valid = true;
+  sect = section_from_bytes<COMP_KMXPLUS_SECT>(rawdata+ex->kmxplus.dpKMXPlus, ex->kmxplus.dwKMXPlusSize, valid);
   if (sect == nullptr) {
-    DebugLog("kmx_plus(): 'sect' did not validate");
+    DebugLog("kmx_plus(): 'sect' missing or did not validate");
     valid = false;
   } else {
-    valid = true;
     // load other sections, validating as we go
-    // each field will be set to nullptr if validation fails
-    bksp = section_from_sect<COMP_KMXPLUS_BKSP>(sect);
-    disp = section_from_sect<COMP_KMXPLUS_DISP>(sect);
-    elem = section_from_sect<COMP_KMXPLUS_ELEM>(sect);
-    key2 = section_from_sect<COMP_KMXPLUS_KEYS>(sect);
-    layr = section_from_sect<COMP_KMXPLUS_LAYR>(sect);
-    list = section_from_sect<COMP_KMXPLUS_LIST>(sect);
-    loca = section_from_sect<COMP_KMXPLUS_LOCA>(sect);
-    meta = section_from_sect<COMP_KMXPLUS_META>(sect);
-    strs = section_from_sect<COMP_KMXPLUS_STRS>(sect);
-    tran = section_from_sect<COMP_KMXPLUS_TRAN>(sect);
-    uset = section_from_sect<COMP_KMXPLUS_USET>(sect);
-    vars = section_from_sect<COMP_KMXPLUS_VARS>(sect);
+    // each field will be set to nullptr if validation fails or if the section is missing
+    bksp = section_from_sect<COMP_KMXPLUS_BKSP>(sect, valid);
+    disp = section_from_sect<COMP_KMXPLUS_DISP>(sect, valid);
+    elem = section_from_sect<COMP_KMXPLUS_ELEM>(sect, valid);
+    key2 = section_from_sect<COMP_KMXPLUS_KEYS>(sect, valid);
+    layr = section_from_sect<COMP_KMXPLUS_LAYR>(sect, valid);
+    list = section_from_sect<COMP_KMXPLUS_LIST>(sect, valid);
+    loca = section_from_sect<COMP_KMXPLUS_LOCA>(sect, valid);
+    meta = section_from_sect<COMP_KMXPLUS_META>(sect, valid);
+    strs = section_from_sect<COMP_KMXPLUS_STRS>(sect, valid);
+    tran = section_from_sect<COMP_KMXPLUS_TRAN>(sect, valid);
+    uset = section_from_sect<COMP_KMXPLUS_USET>(sect, valid);
+    vars = section_from_sect<COMP_KMXPLUS_VARS>(sect, valid);
```

Comment:
And I see this is why you are passing state through. Perhaps it'd be better to do this to keep the functions purer and match the subsequent `setTran` etc functions below?

```
valid = section_from_sect<COMP_KMXPLUS_BKSP>(sect, bksp) && valid;
```

But understand if you prefer to keep it as is.

Reply:
> valid = section_from_sect<COMP_KMXPLUS_BKSP>(sect, bksp) && valid;  

yeah, i could see that being less confusing.  And make missing be still valid.